### PR TITLE
SetGridField fixes

### PR DESCRIPTION
**Pull request by sapirshuker**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-30932
https://jira-dc.paloaltonetworks.com/browse/XSUP-31038
https://jira-dc.paloaltonetworks.com/browse/XSUP-30050

## Description
Fix an issue that the script to raise an uninformative error when trying to run SetGridField from the playground.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/ReleaseNotes/1_12_56.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+#### Scripts
+
+##### SetGridField
+
+- Fix an issue that the script raised an uninformative error when trying to run SetGridField from the playground and from the debugger.
```
Comment (GuyAfik) [inline Packs/CommonScripts/ReleaseNotes/1_12_56.md:6]:
```suggestion
- Fixed an issue that the script raised an uninformative error when trying to run the script from the playground and from the playbook debugger.
```

File: Packs/CommonScripts/Scripts/SetGridField/SetGridField.py [hunk 2]
```diff
@@ -161,8 +161,15 @@ def get_current_table(grid_id: str) -> pd.DataFrame:
...
-    custom_fields = demisto.incident().get("CustomFields", {})
-    current_table: Optional[List[dict]] = custom_fields.get(grid_id)
+    incident = demisto.incident()
+    custom_fields = incident.get("CustomFields", {}) or {}
+    is_playground = incident.get("isPlayground")
+    if is_playground and grid_id not in custom_fields:
+        raise ValueError(f"The following grid id was not found: {grid_id}. Please make sure you entered the correct "
+                         f"incident type with the \"Machine name\" as it appears in the incident field editor in "
+                         f"Settings->Advanced ->Fields (Incident). Also make sure that this value appears in the "
+                         f"incident Context Data under incident - if not then please consult with support.")
```
Comment (GuyAfik) [inline Packs/CommonScripts/Scripts/SetGridField/SetGridField.py:171]:
```suggestion
        raise ValueError(f"The following grid id was not found: {grid_id}. Make sure you entered the correct "
                         f"incident type with the \"Machine name\" as it appears in the incident field editor in "
                         f"Settings->Advanced ->Fields (Incident). Also make sure that this value appears in the "
                         f"incident Context Data under incident - if not then consult with PANW support team.")
```

File: Packs/CommonScripts/Scripts/SetGridField/SetGridField.py [hunk 3]
```diff
@@ -362,24 +369,27 @@ def main():  # pragma: no cover
                                    keys_from_nested=argToList(args.get('keys_from_nested'))
                                    )
         # Execute automation 'setIncident` which change the Context data in the incident
-        res = demisto.executeCommand("setIncident", {
+        res_set = demisto.executeCommand("setIncident", {
```
Comment (GuyAfik) [inline Packs/CommonScripts/Scripts/SetGridField/SetGridField.py:372]: why changing the name?

File: Packs/CommonScripts/Scripts/SetGridField/SetGridField.py [hunk 4]
```diff
@@ -362,24 +369,27 @@ def main():  # pragma: no cover
...
         # we want to check if the incident was succefully updated
         # we execute command and not using `demisto.incident()` because we want to get the updated incident and context
         res = demisto.executeCommand("getIncidents", {"id": demisto.incident().get("id")})
-        data = res[0]["Contents"]["data"]
-        custom_fields = data[0].get("CustomFields")
-
-        if table and grid_id not in custom_fields:
+        is_playground = demisto.incident().get("isPlayground")
+        custom_fields = {}
+        for res_obj in res:
```
Comment (GuyAfik) [inline Packs/CommonScripts/Scripts/SetGridField/SetGridField.py:382]:
```suggestion
        for entry in res:
```

**Review by GuyAfik — COMMENTED** [review]
@sapirshuker LGTM, nice work

File: Packs/CommonScripts/ReleaseNotes/1_12_56.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+
+#### Scripts
+
+##### SetGridField
+
+- Fixed an issue that the script raised an uninformative error when trying to run the script from the playground and from the playbook debugger.
```
Comment (eyalpalo) [inline Packs/CommonScripts/ReleaseNotes/1_12_56.md:6]:
```suggestion
- Fixed an issue where an uninformative error was raised when trying to run the script from the playground or the playbook debugger.
```

File: Packs/CommonScripts/Scripts/SetGridField/SetGridField.py [hunk 6]
```diff
@@ -362,24 +369,27 @@ def main():  # pragma: no cover
...
         # we execute command and not using `demisto.incident()` because we want to get the updated incident and context
         res = demisto.executeCommand("getIncidents", {"id": demisto.incident().get("id")})
-        data = res[0]["Contents"]["data"]
-        custom_fields = data[0].get("CustomFields")
-
-        if table and grid_id not in custom_fields:
-            raise ValueError(f"The following grid id was not found: {grid_id}. Please make sure you entered the correct "
+        is_playground = demisto.incident().get("isPlayground")
+        custom_fields = {}
+        for entry in res:
```
Comment (eyalpalo) [inline Packs/CommonScripts/Scripts/SetGridField/SetGridField.py:382]: Why do we iterating over the incidents? we will only get one, the one with the correct id given.

File: Packs/CommonScripts/Scripts/SetGridField/SetGridField_test.py [hunk 7]
```diff
@@ -189,3 +190,73 @@ def test_build_grid_command_with_multi_sort_by(mocker, keys: list[str], columns:
...
+    mocker.patch.object(SetGridField, 'get_current_table', return_value=[])
+    mocker.patch.object(SetGridField, 'build_grid_command', return_value=[{"name": "name", "readable_name": "readable_name"}])
+    mocker.patch.object(demisto, 'executeCommand', side_effect=side_effect_for_execute_command_in_playground)
+    mocked_return_err = mocker.patch.object(SetGridField, "return_error")
+    SetGridField.main()
+    assert not mocked_return_err.called
+
+
+def test_main_raises_list_index_not_in_playground(mocker):
+    """Unit test
```
Comment (eyalpalo) [inline Packs/CommonScripts/Scripts/SetGridField/SetGridField_test.py:227]:
```suggestion
    """
```

File: Packs/CommonScripts/Scripts/SetGridField/SetGridField_test.py [hunk 8]
```diff
@@ -189,3 +190,73 @@ def test_build_grid_command_with_multi_sort_by(mocker, keys: list[str], columns:
...
+    mocker.patch.object(SetGridField, 'get_current_table', return_value=[])
+    mocker.patch.object(SetGridField, 'build_grid_command', return_value=[{"name": "name", "readable_name": "readable_name"}])
+    mocker.patch.object(demisto, 'executeCommand', side_effect=side_effect_for_execute_command_in_playground)
+    mocker_return_error = mocker.patch('SetGridField.return_error')
+    SetGridField.main()
+    assert mocker_return_error.called
+
+
+def test_get_current_table_exception(mocker):
+    """Unit test
```
Comment (eyalpalo) [inline Packs/CommonScripts/Scripts/SetGridField/SetGridField_test.py:250]:
```suggestion
    """
```

**Review by eyalpalo — CHANGES_REQUESTED** [review]
Nice!

File: Packs/CommonScripts/Scripts/SetGridField/SetGridField.py (hunk 6 above)
Comment (sapirshuker) [reply Packs/CommonScripts/Scripts/SetGridField/SetGridField.py:382]: We do not iterate over the the incident we do iterate over the response, it is an implementation base on Shahaf bug, 
He introduced it in the daily.

File: Packs/CommonScripts/Scripts/SetGridField/SetGridField.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sapirshuker) [reply Packs/CommonScripts/Scripts/SetGridField/SetGridField.py]: We do not iterate over the the incident we do iterate over the response, it is an implementation base on Shahaf bug. He introduced it in the daily.

File: Packs/CommonScripts/Scripts/SetGridField/SetGridField.py [hunk 9]
```diff
@@ -362,24 +382,23 @@ def main():  # pragma: no cover
...
-        if is_error(res):
+        is_playground = demisto.incident().get("isPlayground")
+        custom_fields = {}
+        for entry in res:
+            if entry['Contents']:
+                data = entry["Contents"]["data"]
+                custom_fields = data[0].get("CustomFields") if data and data[0].get("CustomFields") else {}
+        if (not is_playground) and table and grid_id not in custom_fields:
+            raise ValueError(get_error_message(grid_id))
+        if is_error(res_set):
```
Comment (ilaner) [inline Packs/CommonScripts/Scripts/SetGridField/SetGridField.py:401]: let's add `is_error` to the `getIncidents` command as well in line 392

**Review by ilaner — APPROVED** [review]
Good job! see small comment

**Review by eyalpalo — APPROVED** [review]
Looking good! Add what @ilaner asked and merge

**Pull request merged**